x-pack/legacy/plugins/apm/server/routes/index_pattern.ts

```diff
@@ -4,21 +4,32 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import * as t from 'io-ts';
-import { getAPMIndexPattern } from '../lib/index_pattern';
+import { createStaticIndexPattern } from '../lib/index_pattern/create_static_index_pattern';
 import { createRoute } from './create_route';
-import { getKueryBarIndexPattern } from '../lib/index_pattern/getKueryBarIndexPattern';
 import { setupRequest } from '../lib/helpers/setup_request';
 
-export const indexPatternRoute = createRoute(core => ({
-  path: '/api/apm/index_pattern',
-  handler: async () => {
+export const staticIndexPatternRoute = createRoute(core => ({
+  method: 'POST',
+  path: '/api/apm/index_pattern/static',
+  handler: async (req, params, h) => {
     const { server } = core.http;
-    return await getAPMIndexPattern(server);
+    try {
+      await createStaticIndexPattern(server);
+    } catch (e) {
+      const alreadyExists = e.body.error.reason.includes(
```

Comment:
Turns out it's a 409 with a type of `version_conflict_engine_exception`:

https://github.com/elastic/kibana/blob/4c33f2a359b80c3eb63e66312e77ca30410a369c/src/core/server/saved_objects/service/lib/repository.ts#L926-L927

I'll change it to use the type instead of parsing the reason.

Reply:
Turns out there is a helper for handling these errors `SavedObjectsErrorHelpers.isConflictError(e)`

https://github.com/elastic/kibana/blob/master/docs/development/core/server/kibana-plugin-server.savedobjectserrorhelpers.isconflicterror.md